ibis/expr/types/relations.py

```diff
@@ -2417,12 +2441,31 @@ def drop(self, *fields: str | Selector) -> Table:
         │ Torgersen │               193 │        3450 │ female │  2007 │
         └───────────┴───────────────────┴─────────────┴────────┴───────┘
         """
-        if not fields:
+
+        def _get_cols(field) -> list[str]:
+            try:
+                cols = self._fast_bind(field)
+            except com.IbisTypeError as e:
+                if "is not found in table" in str(e):
+                    if missing_ok:
+                        return []
+                    raise
+            except AttributeError as e:
```

Comment:
Needing to catch these two different kinds of exceptions here, and them being not very specific, I think really demonstrates the need for https://github.com/ibis-project/ibis/pull/10412